cli/deno_dir.rs

```diff
@@ -630,74 +701,52 @@ fn fetch_remote_source_async(
         match fetch_once_result {
           FetchOnceResult::Redirect(url) => {
             // If redirects, update module_name and filename for next looped call.
-            let resolve_result = dir
-              .resolve_module(&(url.to_string()), ".")
-              .map_err(DenoError::from);
-            match resolve_result {
-              Ok((new_module_name, new_filename)) => {
-                if maybe_initial_module_name.is_none() {
-                  maybe_initial_module_name = Some(module_name.clone());
-                  maybe_initial_filename = Some(filename.clone());
-                }
-                // Not yet completed. Follow the redirect and loop.
-                Ok(Loop::Continue((
-                  dir,
-                  maybe_initial_module_name,
-                  maybe_initial_filename,
-                  new_module_name,
-                  new_filename,
-                )))
-              }
-              Err(e) => Err(e),
+            let (new_module_name, new_filename) = dir
+              .resolve_module(&(url.to_string()), ".")?;
```

Comment:
`url.as_str()` can you use instead?

Reply:
Nope, `url` is of type `http::uri::Uri` it doesn't have `as_str` method. This was addressed in #2559